### PR TITLE
Implemented show/hide search form functionality

### PR DESCRIPTION
**Pull request by estambakio-sc**
Related issue: #40

### CODE REVIEW

File: config/webpack.config.showroom.js [hunk 1]
```diff
@@ -104,7 +104,7 @@ module.exports = {
       template: '../www/index.html',
     }),
     new webpack.NamedModulesPlugin(),
-    new webpack.optimize.ModuleConcatenationPlugin(),
+    // new webpack.optimize.ModuleConcatenationPlugin(),
```
Comment (amourzenkov-sc) [inline config/webpack.config.showroom.js]: Remove if not needed OR comment the reason for keeping

File: src/crudeditor-lib/views/search/constants.js [hunk 2]
```diff
@@ -40,4 +40,6 @@ export const
 
   VIEW_REDIRECT_REQUEST = namespace + '/VIEW_REDIRECT_REQUEST',
   VIEW_REDIRECT_FAIL = namespace + '/VIEW_REDIRECT_FAIL',
-  VIEW_REDIRECT_SUCCESS = namespace + '/VIEW_REDIRECT_SUCCESS';
+  VIEW_REDIRECT_SUCCESS = namespace + '/VIEW_REDIRECT_SUCCESS',
+
+  TOGGLE_SEARCH_FORM = namespace + '/TOGGLE_SEARCH_FORM';
```
Comment (amourzenkov-sc) [inline src/crudeditor-lib/views/search/constants.js]: Use [redux-actions](https://github.com/reduxactions/redux-actions) naming convention:
action name: <NOUN>_<VERB>
action creator name: <verb><Noun>
selector name: get<Noun>
[source](https://medium.com/@kylpo/redux-best-practices-eef55a20cc72)

File: src/components/SearchMain/index.js [hunk 3]
```diff
@@ -27,32 +33,65 @@ export default class SearchMain extends PureComponent {
 
   render() {
     const { model } = this.props;
+    const {
```
Comment (amourzenkov-sc) [inline src/components/SearchMain/index.js]: For consistency sake, define canCreate here

File: src/components/SearchMain/index.js [hunk 4]
```diff
@@ -27,32 +33,65 @@ export default class SearchMain extends PureComponent {
...
+        <h1>
+          <Row>
+            <Col xs={8}>
+
+              {getModelMessage(i18n, 'model.name')}
+
+              <Button
+                bsStyle="link"
+                onClick={toggleSearchForm}
+                title={`${hideSearchForm ? 'Show' : 'Hide'} search form`}
```
Comment (amourzenkov-sc) [inline src/components/SearchMain/index.js]: Implement title translation

File: src/components/SearchMain/index.js (hunk 3 above)
Comment (estambakio-sc) [reply src/components/SearchMain/index.js]: All done.

**Pull request merged**